pkg/build/controller/factory/factory.go

```diff
@@ -95,6 +96,10 @@ func (factory *BuildControllerFactory) Create() controller.RunnableController {
 			build := obj.(*buildapi.Build)
 			err := buildController.HandleBuild(build)
 			if err != nil {
+				// Set default Reason.
+				if build.Status.Reason == "" {
```

Comment:
Isn't grep losing context in this case?
AFAIK the idiomatic way to check if a string is the empty string in Go is to do that obvious comparison.
https://groups.google.com/d/msg/golang-nuts/7Ks1iq2s7FA/GOujoyeYsOcJ
That post dates back to 2010, and for already some years the compiler is good enough that there's no significant difference in compiled code.